### ISSUE TITLE
Feature request: automatically locate mosaic censoring

### ISSUE THREAD

**Issue by sparr** [deeppomf/DeepCreamPy#27]
The process of duplicating images and then painting one green for mosaics seems like it could be automated. Mosaics have identifiable characteristics and boundaries.

**Comment by EIREXE** [comment]
Not all mosaics are the same, some i've found even seem to not always use pixels.

**Comment by lstama** [comment]
Mosaics characteristics (for censoring purpose in anime or manga) usually are just some _perfect_ squares with different color and same size at each square (grayscale or color mosaics). This kind of mosaics are easy to detect. 

Turn the picture to grayscale, try to find perfect square(s) for each value (from 0 to 255), and then change all the squares to green in the original picture.

But mosaics isn't the only censoring method, sometimes they use god rays, blur, or even emojis.

**Comment by deeppomf** [comment]
Mosaics do have distinct characteristics, but I don't know how to translate them into computer vision terms. The problem is made harder by the bodily fluids covering the mosaic censors.

**Comment by Bobwu1** [comment]
Give me some training data (for research) and I'll look into it. Alternatively, you can just train up a rcnn or mask cnn to detect the mosaics.

**Comment by deeppomf** [comment]
I don't have data for segmentation annotations of mosaics.
Automatic detection of mosaics with RCNN is low priority. The more neural networks I add, the bigger the file size will be.

**Comment by cooldude4232** [comment]
I think that the best way to go about this would just to do square/rectangle detection using something like opencv and then coloring all the squares that have an unusually high occurrence. It's a simple enough problem that you shouldn't need to do neural nets, just need to specify whether or not it is a mosaic

**Comment by Kinomora** [comment]
Danbooru > searchtag: censored
For training material!
